File: payment-service/src/main/java/com/hedvig/paymentservice/serviceIntergration/productPricing/ProductPricingService.java
package com.hedvig.paymentservice.serviceIntergration.productPricing;

import com.hedvig.paymentservice.query.member.entities.Transaction;
import com.hedvig.paymentservice.serviceIntergration.productPricing.dto.InsuranceStatus;
import com.hedvig.paymentservice.serviceIntergration.productPricing.dto.PolicyGuessResponseDto;

import java.time.YearMonth;
import java.util.Collection;
import java.util.Map;
import java.util.Optional;
import java.util.UUID;

public interface ProductPricingService {
  Optional<InsuranceStatus> getInsuranceStatus(String memberId);

  Map<UUID, Optional<PolicyGuessResponseDto>> guessPolicyTypes(Collection<Transaction> transactions, YearMonth period);
}

File: payment-service/src/main/java/com/hedvig/paymentservice/serviceIntergration/productPricing/ProductPricingService.kt
package com.hedvig.paymentservice.serviceIntergration.productPricing

import com.hedvig.paymentservice.query.member.entities.Transaction
import com.hedvig.paymentservice.serviceIntergration.productPricing.dto.ContractMarketInfo
import com.hedvig.paymentservice.serviceIntergration.productPricing.dto.PolicyGuessResponseDto
import java.time.YearMonth
import java.util.Optional
import java.util.UUID

interface ProductPricingService {
    fun guessPolicyTypes(
        transactions: Collection<Transaction>,
        period: YearMonth
    ): Map<UUID, Optional<PolicyGuessResponseDto>>

    fun getContractMarketInfo(memberId: String): ContractMarketInfo

    fun hasContractActiveCurrentMonth(memberId: String): Boolean
}
